### PR TITLE
Allocation improvements and formatting

### PR DESCRIPTION
**Pull request by xperiandri**
I would go like this. There are 2 commits to see changes granularly

### CODE REVIEW

File: src/FSharp.Data.GraphQL.Shared/Errors.fs [hunk 1]
```diff
@@ -145,52 +146,60 @@ type GQLProblemDetails = {
...
+    static member Create (message, [<Optional>] extensions) = {
         Message = message
         Exception = ValueNone
         Path = Skip
         Locations = Skip
         Extensions = extensions
     }
 
-    static member CreateOfException (message : string, ex : Exception, ?path : FieldPath, ?extensions : IReadOnlyDictionary<string, obj>) = {
+    static member CreateOfException (message : string, ex : Exception, [<Optional>] path : FieldPath Skippable, [<Optional>] extensions : IReadOnlyDictionary<string, obj> Skippable) = {
```
Comment (valbers) [inline src/FSharp.Data.GraphQL.Shared/Errors.fs:165]: I tried to use the `[<Optional>]` attribute as you suggested, but it wasn't being recognized. So it is in `System.Runtime.InteropServices` as it seems...

**Comment by valbers** [comment]
@xperiandri thank you. Regarding the formatting, what do you think about simply reverting my commit with the formatting changes (it was in one specific commit) and then I see how I can fix my fantomas configuration (it's a good idea for future changes)?

**Comment by xperiandri** [comment]
If you hard reset to my commit it will already be correct. Or revert as you propsed
image

**Review by valbers — APPROVED** [review]
LGTM

**Pull request merged**